app/models/campaign.rb

```diff
@@ -10,4 +10,8 @@ class Campaign < ApplicationRecord
                         numericality: true,
                         length: { is: 16 },
                         format: { with: /\A\d+\z/ }
+
+  def percent_of_collected_amount
+    needed_amount.zero? ? 0 : 100 * current_amount / needed_amount
```

Comment:
`100 * current_amount / needed_amount` looks like `(100*4)/20` 
Is it correct?

Reply:
Yes. (100*4)/20 = 4/20*100. In math the result will be the same.
But in ruby: 4/20 = 0 and 0*100 = 0%
It's correct:  100*4 = 400 and 400/20 = 20%

Reply:
Maybe if needed amount is 0, you should return 100%
If needed is 0 and we have 0, then we collected all the money we needed :)